src/Components/BookPage/BookSynopsis.js

```diff
@@ -1,65 +1,467 @@
+    // Kindergarten to short grade
+    gradeRange = `K - ${gradeMax}`;
+  } else if (gradeMin === '0 to Pre-K' && gradeMax === 'Kindergarten') {
+    // 0 to Pre-K to Kindergarten
+    gradeRange = '0 to K';
+  } else if (gradeMin === gradeMax) {
+    // gradeMin same as gradeMax
+    gradeRange = gradeMin;
+  } else {
+    // Short grade range e.g. 3rd to 6th
+    gradeRange = `${gradeMin} to ${gradeMax}`;
+  }
+  return gradeRange;
+};
+
+/* Age Range
+    An ageRange will render regardless of whether or not data is good
+  */
+const createAgeRange = (ageMin, ageMax) => {
+  let ageRange;
+  if (ageMin === -1 && ageMax === -1) {
+    // Bofa
+    ageRange = 'All';
+  } else if (ageMin === ageMax) {
+    // Valid but equal ages
+    ageRange = `Age ${ageMin}`;
+  } else if (ageMin === -1) {
+    // Bad ageMin
+    ageRange = `0 - ${ageMax}`;
+  } else if (ageMax === -1) {
+    // Bad ageMax
+    ageRange = `${ageMin} - 18`;
+  } else if (ageMax < ageMin) {
+    // Bad ordering
+    ageRange = `${ageMax} - ${ageMin}`;
+  } else {
+    // Happy trail
+    ageRange = `${ageMin} - ${ageMax}`;
+  }
+  return ageRange;
+};
+
+function BookSynopsis({
+  title, authorName, authorID, illustratorName, illustratorID, desc, imageURL,
+  bookshopURL, readAloudURL, identityTags, ageMin, ageMax, gradeMin, gradeMax,
+}) {
+  const [seeMore, setSeeMore] = useState(true);
+  const toggleSeeMore = () => setSeeMore(!seeMore);
+
+  const identityChips = identityTags.map((tag) => {
+    /* Randomly select colors for identity tag
+       This does not guarantee a variety of colors.
+    */
+    const rand = Math.floor(Math.random() * 3);
+    const chipStyle = { ...chipColors[rand], ...styles.chip };
+    return (
+      <Chip
+        label={tag}
+        sx={chipStyle}
+        color="primary"
+        key={tag}
+        id={tag}
+      />
+    );
+  });
+
+  const gradeRange = createGradeRange(gradeMin, gradeMax);
+  const ageRange = createAgeRange(ageMin, ageMax);
+  return (
+    <Box sx={styles.synopsis}>
+      <Box sx={styles.topDownContainer}>
+        <Box
+          component="img"
+          sx={styles.bookCover}
+          alt={`Book cover for ${title}`}
+          src={imageURL}
+        />
+        <Typography sx={styles.tagHeader}>Tags</Typography>
+        <Box sx={styles.tagContainer}>
+          {identityChips}
+        </Box>
+      </Box>
+
+      <Card sx={styles.description}>
+        <CardContent>
+          <Box>
+            <Typography sx={styles.bookTitle}>{title}</Typography>
+          </Box>
+
+          <Box variant="body2">
+            {
+              (desc.length > 525) ? (
+                <Box>
+                  <Typography sx={styles.text}>
+                    {
+                      (seeMore) ? `${desc.substring(0, 525)}...` : desc
+                    }
+                  </Typography>
+                  <Box sx={styles.seeMoreContainer} onClick={toggleSeeMore}>
+                    <Typography sx={styles.seeMoreText}>
```

Comment:
For some reason, the text and icon of this button highlight separately on hover. This is fine, but it is a little confusing, so if you can fix it that would be helpful!